### PR TITLE
Re readable body

### PR DESCRIPTION
**Pull request by robinedman**
## What's new
Changed buttons on list page and added _Preview_ button:
screen shot 2015-02-23 at 17 53 21

New _Preview_ page:
screen shot 2015-02-23 at 17 44 24

Note that there's a separate PR for organising the JS code better. In the interests of small PRs and all that. 

## QA
Create an article with different kinds of elements. Also add some embed and more text after it.
 
- Test that it's still possible to restore.
- Test copying from the restorer preview to a Composer article. Include some notes.

### CODE REVIEW

**Comment by rrees** [comment]
:+1:

**Comment by robinedman** [comment]
Since these two PRs both have :+1:'s now I think it makes sense to close this one in favour #40 . To avoid unneccessary double QA work.

File: app/assets/javascripts/show-version.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * Version model
+ */
+
+function isTextElement(element) {
+    return element.elementType === 'text';
+}
+
+
+function getTextFields(version) {
+    function extractTextField(element) {
```
Comment (cutandpastey) [inline app/assets/javascripts/show-version.js]: Why is this nested and `isTextElement` is not?

File: app/assets/javascripts/show-version.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ */
+
+function renderVersion() {
+    var versionPath = getJSONVersionPath();
+    request.get(versionPath, function(error, response) {
+        var version = JSON.parse(response.text);
+
+        var textFields = getTextFields(version);
+
+        var versionTemplate = require('./version.handlebars');
```
Comment (cutandpastey) [inline app/assets/javascripts/show-version.js]: nit-pick but shouldn't this be at the top (just for clarity)?

File: app/assets/javascripts/show-version.js (hunk 1 above)
Comment (robinedman) [inline app/assets/javascripts/show-version.js]: Good catch!

File: app/assets/javascripts/show-version.js (hunk 2 above)
Comment (robinedman) [inline app/assets/javascripts/show-version.js]: Agree that's better.

I'm putting the fixes in #40 and closing this one to keep things simple.

File: app/assets/javascripts/index.js [hunk 3]
```diff
@@ -106,3 +109,36 @@ function enabledIfAllChecked() {
...
+        name: 'Show readable version',
+        pattern: 'versions/version/readable',
+        action: showVersionModule
+    }
+];
+
+// Uses Array.prototype.some to only find the first matching route.
+routes.some(function(route) {
+    // TODO: Use more proper router. Just a very simple matching algorithm atm.
+    if (window.location.pathname.indexOf(route.pattern) !== -1) {
```
Comment (cutandpastey) [inline app/assets/javascripts/index.js]: You should checkout https://github.com/jashkenas/backbone/blob/master/backbone.js#L1381 for inspiration on this one.

File: public/stylesheets/application.css [hunk 4]
```diff
@@ -390,4 +390,33 @@ html {
...
+
+gu-correct {
+    background-color: greenyellow;
+}
+
+.version-body {
+    /* Remove default headline */
+    outline: none;
+
+    margin-top: 5vh;
```
Comment (cutandpastey) [inline public/stylesheets/application.css]: :+1: for `vh` && `vw`! Had to look at what these were :)

**Pull request closed without merging**